### ISSUE TITLE
Improve look/feel of premium block promos on WordPress.com

### ISSUE THREAD

**Issue by davemart-in** [Automattic/jetpack#15190]
We've got a hand full of blocks on WP.com that require a specific plan. If you're on a lower plan, you'll see a prompt to upgrade in order to use these blocks. The design of these promo's could use some love.

#### Steps to reproduce the issue

1. Create a WordPress.com free account.
2. Add a page
3. Add one of the following blocks:

- Simple Payments
- Recurring Payments
- OpenTable
- Calendly

#### What I expected
Promo looks consistent, minimal, and well designed.

#### What happened instead

**Screenshots**

Here's what these promos look like now:

<img width="770" alt="simple-payments" src="https://user-images.githubusercontent.com/5634774/77906877-b35a1280-7256-11ea-8dae-4ae8c2261c53.png">

<img width="808" alt="recurring-payments" src="https://user-images.githubusercontent.com/5634774/77906886-b6ed9980-7256-11ea-9878-883d6c7d61d2.png">

<img width="779" alt="opentable" src="https://user-images.githubusercontent.com/5634774/77906894-ba812080-7256-11ea-95a1-063846f31e93.png">

<img width="769" alt="calendly" src="https://user-images.githubusercontent.com/5634774/77906901-be14a780-7256-11ea-9adf-a1ebc73be47e.png">

**Comment by davemart-in** [comment]
@olesyabrk shared some design concepts for this:

A) 
![image-40](https://user-images.githubusercontent.com/5634774/77907475-b275b080-7257-11ea-9aa6-0cafe5f43438.png)

B) 
![image-41](https://user-images.githubusercontent.com/5634774/77907494-b6a1ce00-7257-11ea-91b6-63a7fbdb1680.png)

C)
![image-42](https://user-images.githubusercontent.com/5634774/77907508-bb668200-7257-11ea-8fb0-7f6e4b6360f8.png)

D) 
![image-43](https://user-images.githubusercontent.com/5634774/77907520-c02b3600-7257-11ea-842e-3985eba3f775.png)

**Comment by keoshi** [comment]
Are these actual screenshots of the upgrade banners?

**Comment by davemart-in** [comment]
The ones in the issue are, yes. They're pretty inconsistent and could use some love.

**Comment by keoshi** [comment]
That's really odd. Could be caused by the latest styling changes on G2 and it almost seems like some classes aren't even being targeted anymore. I have a slight preference for proposal A and I'd love to see this fixed in a couple of days.

@folletto @jeherve so you're aware of this.

**Comment by folletto** [comment]
I agree on proposal A, but... I'd like to ping @jasmussen to see if he has any advice on how to style a block like that in the new G2 world.

**Comment by davemart-in** [comment]
My feedback on option A: It's important that they're there, but we don't really need to draw too much attention to these promos. 

**Button**
We already use a primary button in some of the blocks now, so we should probably stick with secondary button styling for the "Upgrade" button. 

**Border**
The other thing I wonder about is the need for the solid dark border around the promo.

**Comment by folletto** [comment]
Yes I agree. I'd also note that the example of that specific block is a bit misleading as it has a lot going on inside too, which doesn't help in finding the right solution. I'd try to find a solution first that fits well the paradigm, and then try and apply to 2-3 blocks and see how it works.

**Comment by keoshi** [comment]
One more thing to bear in mind is that right now in all (most?) contexts the bounding box/outline on the mocks doesn't exist on the block, so the separation between elements and blocks might appear more arbitrary than what we're seeing here.

**Comment by jasmussen** [comment]
Thanks for the ping! Excited about premium blocks.

Here's a GIF showing the current block as I see it:

![gif](https://user-images.githubusercontent.com/1204802/77914393-246cea80-7296-11ea-902e-d83c799e9809.gif)

Separate to the conversation on notices specifically, and I'll circle back to that in a moment, I would love to see an effort made to reduce the jump that happens from selected to unselected states. It is, in some cases, unavoidable. But in the noble effort to try and avoid that, it's possible that the placeholder state can move closer to the final state, so the switch isn't so jarring. It isn't inherently wrong — but the placeholder state doesn't _have_ to look like a barren form field. 

Imagine the placeholder state looking almost exactly like the final rendered state when you first insert it, and then you click it to reveal that every part of it is customizable, without a shift in dimensions:

<img width="518" alt="Frame 120" src="https://user-images.githubusercontent.com/1204802/77915288-7feba800-7297-11ea-8d33-5977d6451dc7.png">

This is something the recurring payments block handles already:

![recurring](https://user-images.githubusercontent.com/1204802/77915395-b1647380-7297-11ea-84be-b81ff729b3da.gif)

That's obviously a separate effort to the question of the notice, but worth noting nevertheless.

For the notice, I have only a few thoughts on the treatment:

- Because it's not actually a WordPress component, it should not have to be beholden to any G2 design. It could make sense to leverage DNA, like icon style, inherited fonts, metrics and so on, just to feel like it belongs in the context. But I wanted to just clarify that if for whatever reason you'd prefer to put a powerful WordPress.com blue as the background with white text on it, no G2 efforts should hold you back — only a sense of good design!
- If you do want to be totally beholden to some of the G2 elements that are currently trickling out, consider using (and improving!) [the actual Notice component](https://github.com/WordPress/gutenberg/tree/master/packages/components/src/notice). It's not in Storybook, the closest equivalent is [the snackbar](https://wordpress.github.io/gutenberg/?path=/story/components-snackbar--with-actions), and in fact the Notice component is so unused it could actually use a lot of love to help prove its worth and make it part of storybook.
- I'd avoid drop shadows here, because presumably the notice card lays flat on top of the block, as opposed to floating over it. 

Hope this was helpful!

**Comment by davemart-in** [comment]
Feedback from @mtias:

> I see a few problems with the wrapper element: the selected state seems to add quite a bit of padding around the block content itself. That would mean every time the block is selected there would be a “jump” as the content shrinks to accommodate the padding unless this is how the block is presented all the time, which has its own issues as it would depart a bit from the WYSIWYG principle.

> Generally we have a pretty clear distinction between placeholder cards and the regular content. This approach seems to mix the two (showing the block name and icon in the canvas of the block while the block is showing its actual content). I know the simple payments block is not the best example since it works currently like an embedded form and not like most other blocks out there. However, this mixed effect would likely be more pronounced on the other blocks that need this treatment (recurring payments is just a single button element, for example).

> I’d suggest perhaps looking at adding this notice to the toolbar itself as a possible alternative since it’s not taller than before and should accommodate copy better.

**Comment by pablohoneyhoney** [comment]
If leveraging G2 language, I don't think those explorations feel native. I wonder if this could be solved with these: https://github.com/WordPress/gutenberg/issues/20591

**Comment by davemart-in** [comment]
I removed this from the Earn board. I know that @dwolfe is doing a sprint on this right now.
